extensions/vertx-mdc/latebound-mdc-provider/src/main/java/io/quarkus/vertx/mdc/provider/LateBoundMDCProvider.java

```diff
@@ -0,0 +1,97 @@
+package io.quarkus.vertx.mdc.provider;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.jboss.logmanager.MDCProvider;
+
+/**
+ * Class enabling Quarkus to instantiate a {@link org.jboss.logmanager.MDCProvider}
+ * and set a delegate during runtime initialization.
+ *
+ * While no delegate is set it serves as a NOP MDC.
+ */
+public class LateBoundMDCProvider implements MDCProvider {
+    private static MDCProvider delegate;
```

Comment:
This likely should be declared `volatile`, otherwise the write might not be observed by other threads right away.